cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerTwillRunnable.java

```diff
@@ -115,6 +118,7 @@ protected void configure() {
           bind(DiscoveryServiceClient.class)
               .toProvider(
                   new SupplierProviderBridge<>(masterEnv.getDiscoveryServiceClientSupplier()));
+          bind(AuditLogWriter.class).to(MessagingAuditLogWriter.class).in(Scopes.SINGLETON);
```

Comment:
Any reason it's inside `masterEnv != null`? Should we have something in the other part of if or move it outside?

Reply:
Moving it out.